sql/core/src/main/java/org/apache/spark/sql/sources/v2/reader/SupportsScanColumnarBatch.java

```diff
@@ -30,21 +30,21 @@
 @InterfaceStability.Evolving
 public interface SupportsScanColumnarBatch extends DataSourceV2Reader {
   @Override
-  default List<ReadTask<Row>> createReadTasks() {
+  default List<DataReaderFactory<Row>> createDataReaderFactories() {
```

Comment:
We shall create only one `DataReaderFactory`, and have that create multiple data readers.

Reply:
`DataReaderFactory` is responsible to do serialization and initialize the actual data readers, so data reader creation must be done at executor side, and before that we need to determine how many RDD partitions we want, which is this method doing.

Reply:
I see, make sense. Maybe update the comment to explain why we need a list of `DataReaderFactory`s?

Reply:
I think already did, in `DataSourceV2Reader`

Reply:
I would also mention this:
```
 data reader creation must be done at executor side
```
so it makes it more clear why we shall have a list of `DataReaderFactory`s, but it's only my personal opinions, it's totally fine if we don't include the implementation details in the comment.

Reply:
We mentioned it in the classdoc of `DataReaderFactory`